From 1fff8a266afb5ad0156469add2c984ccfd1303d0 Mon Sep 17 00:00:00 2001
From: Madhu <enometh@net.meer>
Date: Mon, 23 May 2022 20:52:42 +0530
Subject: [PATCH 1/8] xpra/server/source/webcam_mixin.py:
 (WebCamMixin.process_webcam_frame): avoid assert error

Traceback (most recent call last):  File
"/usr/lib/python3.9/site-packages/xpra/server/source/webcam_mixin.py",
line 175, in process_webcam_frame/   assert encoding in
self.webcam_encodings, "invalid encoding specified: %s (must be one of
%s)" % (encoding, self.webcam_encodings)/ AssertionError: invalid
encoding specified: b'jpeg' (must be one of ['png', 'jpeg', 'webp'])
---
 xpra/server/source/webcam.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/xpra/server/source/webcam.py b/xpra/server/source/webcam.py
index 8df256ffb..8bee8e4f1 100644
--- a/xpra/server/source/webcam.py
+++ b/xpra/server/source/webcam.py
@@ -179,6 +179,12 @@ class WebcamMixin(StubSourceMixin):
         try:
             # pylint: disable=import-outside-toplevel
             from xpra.codecs.pillow.decoder import open_only
+            if (str(encoding) == encoding):
+                pass
+            else:
+                tmp_encoding = encoding
+                encoding = encoding.decode("utf-8")
+                print('MADHU: converted %s from encoding=%s SET=%s' % (encoding, tmp_encoding, self.webcam_encodings))
             if encoding not in self.webcam_encodings:
                 raise ValueError(f"invalid encoding specified: {encoding} (must be one of {self.webcam_encodings})")
             rgb_pixel_format = "BGRX"       #BGRX
-- 
2.49.0.9.gd50a5e8939.dirty

